crates/loader/src/local.rs

```diff
@@ -390,6 +389,17 @@ impl LocalLoader {
     }
 }
 
+#[cfg(not(windows))]
+fn safe_canonicalize(path: &Path) -> std::io::Result<PathBuf> {
```

Comment:
Can we add docs to one of these functions so there's a bit more context on why this is needed?